Improve CI speed and attempt to fix flaky tests

psibi opened the pull request:
Two improvements:

- Download the binary of fd and ripgrep. This IMO saves us ~ 3 to 4 minutes.
- My attempt at fixing flaky tests.

brotzeit commented:
Can you please squash the last 3 commits.

samhedin reviewed (approved):
Thank you! Other than what brotzeit said, this looks great :)

psibi commented:
Thanks for the reviews! I have squashed those three commits. But it led to more test failures and I have fixed them in subsequent commits. I'm not sure if it completely fixes things, but it seems to be a improvement.

Overall I will still be keeping an eye on the tests. But I think abstracting the `sit-for` workaround under a single helper function is helpful as updating our workaround in future will reflect everywhere this is used.

brotzeit commented on test/test-helper.el:
```diff
@@ -123,3 +123,9 @@ list of substrings of `STR' each followed by its face."
       (setq-local compilation-ask-about-save nil)))
 (add-hook 'rustic-mode-hook 'rustic-mode-auto-save-hook)
+
+(defun rustic-test--wait-till-finished (buffer)
```
And btw, please don't put too much effort into making existing tests look better, I think it's not worth the time.

psibi replied:
> Maybe we should pass the process to the function since we already got it ?

We don't get in some places. Eg:

- https://github.com/psibi/rustic/blob/1b8e54d76e1364e2dfa9876eb06fdb604281434d/test/rustic-window-test.el#L10
- https://github.com/psibi/rustic/blob/1b8e54d76e1364e2dfa9876eb06fdb604281434d/test/rustic-window-test.el#L25

> And btw, please don't put too much effort into making existing tests look better, I think it's not worth the time.

This whole PR started out as fixing one of the flaky test and I just went ahead with fixing other flaky tests. :-) I re-ran the CI multiple times and it seems to have become stable.

brotzeit replied:
Ok nevermind.

psibi replied:
Thank you!
